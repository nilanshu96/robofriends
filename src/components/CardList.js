import React from 'react';
import Card from './Card';

const CardList = ({robots}) => {
    
    return (
        <div>
            {
                robots.map(user => {
                    return (<Card key={user.id} id={user.id} name={user.name} email={user.email} />);
                })
            }
        </div>
    );
}

export default CardList;