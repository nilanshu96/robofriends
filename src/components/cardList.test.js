import { render } from "@testing-library/react";
import CardList from "./CardList";

test("renders correctly", () => {
    const mockRobots = [{
        id: 1,
        name: "Leanne Graham",
        username: "Bret",
        email: "Sincere@april.biz"
    }]
    const { container } = render(<CardList robots={mockRobots} />);

    expect(container.innerHTML).toMatchSnapshot();
})