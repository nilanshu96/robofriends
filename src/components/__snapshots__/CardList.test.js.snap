// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div><div data-testid=\\"robocard\\" class=\\"tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5\\"><img src=\\"https://robohash.org/1.png?size=200x200\\" alt=\\"robot\\"><div><h2>Leanne Graham</h2><p>Sincere@april.biz</p></div></div></div>"`;
